flink-connector-kafka/src/test/java/org/apache/flink/connector/kafka/sink/KafkaSinkITCase.java

```diff
@@ -606,29 +682,21 @@ public void initializeState(FunctionInitializationContext context) throws Except
     private static final class InfiniteIntegerSource
             implements SourceFunction<Long>, CheckpointListener, CheckpointedFunction {
```

Comment:
Nit: Does this source still need to be a `CheckpointedFunction`? Most of the logic moved to the mapper.